peekingduck/pipeline/nodes/model/yolact_edgev1/yolact_edge_files/model.py

```diff
@@ -0,0 +1,653 @@
+
+    def __call__(self, predictions: Dict[str, torch.Tensor]) -> List[Any]:
+        """
+        Args:
+            loc_data: (tensor) Loc preds from loc layers
+                Shape: [batch, num_priors, 4]
+            conf_data: (tensor) Shape: Conf preds from conf layers
+                Shape: [batch, num_priors, num_classes]
+            mask_data: (tensor) Mask preds from mask layers
+                Shape: [batch, num_priors, mask_dim]
+            prior_data: (tensor) Prior boxes and variances from priorbox layers
+                Shape: [num_priors, 4]
+            proto_data: (tensor) If using mask_type.lincomb, the prototype masks
+                Shape: [batch, mask_h, mask_w, mask_dim]
+        Returns:
+            output of shape (batch_size, top_k, 1 + 1 + 4 + mask_dim)
+            These outputs are in the order: class idx, confidence, bbox coords, and mask.
+            Note that the outputs are sorted only if cross_class_nms is False
+        """
+        loc_data = predictions["loc"]
+        conf_data = predictions["conf"]
+        mask_data = predictions["mask"]
+        prior_data = predictions["priors"]
+        proto_data = predictions["proto"] if "proto" in predictions else None
+
+        out = []
+        batch_size = loc_data.size(0)
+        num_priors = prior_data.size(0)
+        conf_preds = (
+            conf_data.view(batch_size, num_priors, self.num_classes)
+            .transpose(2, 1)
+            .contiguous()
+        )
+
+        for batch_idx in range(batch_size):
+            decoded_boxes = decode(loc_data[batch_idx], prior_data)
+            result = self.detect(batch_idx, conf_preds, decoded_boxes, mask_data)
+            if result is not None and proto_data is not None:
+                result["proto"] = proto_data[batch_idx]
+            out.append(result)
+        return out
+
+    def detect(  # pylint: disable=too-many-arguments
+        self,
+        batch_idx: int,
+        conf_preds: torch.Tensor,
+        decoded_boxes: torch.Tensor,
+        mask_data: torch.Tensor,
+    ) -> Optional[Dict[str, Any]]:
+        """Perform nms for only the max scoring class that isn't background (class 0)"""
+        cur_scores = conf_preds[batch_idx, 1:, :]
+        conf_scores, _ = torch.max(cur_scores, dim=0)
+        keep = conf_scores > self.conf_thresh
+        scores = cur_scores[:, keep]
+        boxes = decoded_boxes[keep, :]
+        masks = mask_data[batch_idx, keep, :]
+
+        if scores.size(1) == 0:
+            return None
+
+        boxes, masks, classes, scores = self.fast_nms(
+            boxes, masks, scores, self.nms_thresh, self.top_k
+        )
+
+        return {"box": boxes, "mask": masks, "class": classes, "score": scores}
+
+    @classmethod
+    def fast_nms(  # pylint: disable=too-many-arguments, bad-classmethod-argument
+        self,
+        boxes: torch.Tensor,
+        masks: torch.Tensor,
+        scores: torch.Tensor,
+        iou_threshold: float = 0.5,
```

Comment:
Why is this parameter named as `iou_threshold` when the caller above (line 609) uses `nms_thresh`?

Reply:
This issue still has issues: below screenshot from my latest `git pull` updates. The caller (line 645) uses `iou_threshold`, the method (line 655) uses `nms_threshold`, the docstrings (line 665) uses `iou_threshold`.

Please fix this.

![image](https://user-images.githubusercontent.com/13075447/182059832-0e572f88-9e0a-40ab-b777-cec2aa30a76f.png)

Reply:
The authors used `nms_threshold` to mean IoU threshold. Keeping this to `iou_threshold` for consistency with other PeekingDuck models.